servicetalk-router-utils-internal/src/main/java/io/servicetalk/router/utils/internal/RouteExecutionStrategyUtils.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.router.utils.internal;
+
+import io.servicetalk.router.api.NoOffloadsRouteExecutionStrategy;
+import io.servicetalk.router.api.RouteExecutionStrategy;
+import io.servicetalk.router.api.RouteExecutionStrategyFactory;
+import io.servicetalk.transport.api.ExecutionStrategy;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+import static java.util.Arrays.stream;
+
+/**
+ * Utilities to handle {@link RouteExecutionStrategy} annotation.
+ */
+public final class RouteExecutionStrategyUtils {
```

Comment:
I am assuming that code here is just refactored from existing jersey classes, so haven't reviewed this here. Is there anything specific you want me to look at?